### PR TITLE
T05 add edit todo

### PR DESCRIPTION
**Pull request by APaltrow**
Task: T-05: Add the possibility to edit item

-Add "pencil" icon to the right corner of item
-When "pencil" icon is clicked show modal with the possibility to edit current item data
-Modal should have save and cancel buttons

### CODE REVIEW

File: src/hooks/useTodoTask.ts [hunk 1]
```diff
@@ -72,6 +72,12 @@ export const useTodoTask = () => {
   const onDeleteTodo = (todoId: number) => {
     deleteTodo(todoId);
   };
+  const onEditTodo = (todoToEdit: ITodo) => {
+    setTodo(todoToEdit);
+  };
+  const onSaveEditTodo = () => {
+    editTodo(todo);
+  };
```
Comment (Yuliia143) [inline src/hooks/useTodoTask.ts]: Use one style of code - indents between functions

File: src/redux/reducers/todoReducer.ts [hunk 2]
```diff
@@ -37,6 +37,20 @@ export const todoReducer = (
...
+      );
+
+      if (!todo) return state;
+
+      const todoIndex = state.todoList.indexOf(todo);
+
+      state.todoList.splice(todoIndex, 1, { ...action.payload });
+
+      return { ...state };
+    }
```
Comment (Yuliia143) [inline src/redux/reducers/todoReducer.ts]: Try to simplify

File: src/hooks/useTodoTask.ts (hunk 1 above)
Comment (APaltrow) [reply src/hooks/useTodoTask.ts]: Added indents between functions.

File: src/redux/reducers/todoReducer.ts (hunk 2 above)
Comment (APaltrow) [reply src/redux/reducers/todoReducer.ts]: Refactored (simplified) logic for EDIT_TODO action.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**